pettingzoo/test/api_test.py

```diff
@@ -439,10 +443,13 @@ def test_action_flexibility(env):
         obs, reward, terminated, truncated, info = env.last()
         if terminated or truncated:
             action = None
-        elif isinstance(obs, dict) and "action_mask" in obs:
-            action = env.action_space(agent).sample(obs["action_mask"])
         else:
-            action = 0
+            mask = (
```

Comment:
I can understand how this works but I feel like it's more clear to do if else rather than the one liner (but I recall someone saying that for testing we don't want super nested things, which I guess is fair)

Reply:
I fixed this in a different way in my code:
```
        if terminated or truncated:
            action = None
        elif isinstance(prev_observe, dict) and "action_mask" in prev_observe:
            action = env.action_space(agent).sample(prev_observe["action_mask"])
        elif "action_mask" in info:
            action = env.action_space(agent).sample(info["action_mask"])
        else:
            action = env.action_space(agent).sample()
```

I feel like my way makes it much simpler to see what's going on, and is consistent with how the rest of the codebase looks. There's an argument to change the tutorials and rest of the codebase to use get() and explain in the documentation maybe that get returns None if the key doesn't exist, but otherwise I feel like we should do it my way. This code makes it clear that obs can be something other than a dict, but info has to be a dict, and that action mask can be stored in one or the other.

Reply:
Sure, this makes sense. Changed!